mailbag/derivatives/warc.py

```diff
@@ -0,0 +1,71 @@
+import os
+from structlog import get_logger
+import mailbag.helper as helper
+from warcio.capture_http import capture_http
+from warcio import WARCWriter
+import requests  # requests *must* be imported after capture_http
+import _thread
+import http.server
+import socketserver
+
+log = get_logger()
+
+from mailbag.derivative import Derivative
+
+
+class WarcDerivative(Derivative):
+    derivative_name = 'warc'
+    derivative_format = 'warc'
+
+    def __init__(self, email_account, **kwargs):
+        log.debug("Setup account")
+        super()
+        
+        self.args = kwargs['args']
+        mailbag_dir = kwargs['mailbag_dir']
+        self.warc_dir = os.path.join(str(mailbag_dir),'warc')
+
+        if not self.args.dry_run:
+            os.makedirs(self.warc_dir)
+
+            self.httpd = []
+            _thread.start_new_thread(helper.startServer,(self.args.dry_run,self.httpd,5000))
```

Comment:
Looking around, it seems like `_thread` might be deprecated and we should use `threading`?